### PR TITLE
[infra-proxy-service] Environment recipes list API

### PR DESCRIPTION
**Pull request by vsingh-msys**
### :nut_and_bolt: Description: What code changed, and why?

- In the shared model, all recipes available in the system are required on the fly to segregate between, available & used recipes. In this environment based Chef Infra Server API to fulfill this requirement.

### :chains: Related Resources
Fixes: #4453 

### :+1: Definition of Done

- Infra environment based recipes list API have been added.

### :athletic_shoe: How to Build and Test the Change

Steps to build:
- `rebuild components/automate-gateway`
- `rebuild components/infra-proxy-service`

Steps to test: 

- To add data using 3rd internal steps https://github.com/chef/automate/blob/master/dev-docs/adding-data/adding_test_data.md#adding-data-to-infra-views [updated]

- gRPC GetEnvironmentRecipes:

```
chef-automate dev grpcurl infra-proxy-service chef.automate.domain.infra_proxy.service.InfraProxyService.GetEnvironmentRecipes -- -d '{"server_id": "chef-server-dev-test", "org_id": "chef-org-dev", "name": "_default"}'

{
  "recipes": [
    "aix::nim_master_setup",
    "aix::nim_master_setup_standalone",
    "chef-client",
  ]
}
```

- HTTP recipes list API:
```
curl -sSkH "api-token: $(get_admin_token)" https://a2-dev.test/api/v0/infra/servers/chef-server-dev-test/orgs/chef-org-dev/environments/_default/recipes?pretty


{
  "recipes": [
    "aix::nim_master_setup",
    "aix::nim_master_setup_standalone",
    "chef-client",
  ]
}
```

### :white_check_mark: Checklist

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [ ] Tests added/updated?
- [ ] Docs added/updated?
- [x] All commits have been signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### :camera: Screenshots, if applicable
Signed-off-by: Vivek Singh <vivek.singh@msystechnologies.com>

### CODE REVIEW

File: go.mod
Comment (vsingh-msys) [inline go.mod:17]: observe `make revendor` changes

**Review by msorens — COMMENTED** [review]
Not able to test. Per your instructions I loaded sample data with `infra_service_load_sample_data` but then got this:
```
[10][default:/src:0]# chef-automate dev grpcurl infra-proxy-service chef.automate.domain.infra_proxy.service.InfraProxyService.GetEnvironmentRecipes -- -d '{"server_id": "chef-server-dev-test", "org_id": "chef-org-dev", "name": "_default"}'
ERROR:
  Code: NotFound
  Message: no server found with ID "chef-server-dev-test"
UnknownError: exit status 1
```

File: .license_scout.yml
Comment (vsingh-msys) [reply .license_scout.yml:77]: I have gone through this file and found that we have allowed certain licenses https://github.com/chef/automate/blob/b777127dacfecf2053092eb728bff879d9ac18be/.license_scout.yml#L60, which we are putting as license_id. so thought of following the same.

If we can skip these entries to `allowed_licenses` I am okay with it? Thanks!

File: api/external/infra_proxy/infra_proxy.proto
Comment (vsingh-msys) [reply api/external/infra_proxy/infra_proxy.proto:214]: I found `GetEnvironmentRecipes` would be more suitable as these recipes fall under the environment namespace ie. `/environments/NAME/recipes` Ref https://docs.chef.io/api_chef_server/#environmentsnamerecipes.

Also, we can have a cookbook based recipes list, `/cookbooks/_recipes` Ref https://docs.chef.io/api_chef_server/#cookbooks_recipes.

**Comment by vsingh-msys** [comment]
> Not able to test. Per your instructions I loaded sample data with infra_service_load_sample_data but then got this:

@msorens I have updated the steps to test with more clarity, it's working only with realistic chef server data which is provided at step 3rd.

**Comment by vsingh-msys** [comment]
Accidentally hit the closed button reopen it again

**Comment by vsingh-msys** [comment]
@msorens I have added the comments, nice to have a look at it? thanks!

File: api/external/infra_proxy/response/environments.proto
Comment (mjingle) [inline api/external/infra_proxy/response/environments.proto:34]: @vsingh-msys Is this list for _Recipes_ or _Environment Recipes_? This detail should be clarified.

Comment (vsingh-msys) [reply api/external/infra_proxy/response/environments.proto:34]: It is Environment Recipes list will update it thanks!

**Review by mjingle — APPROVED** [review]
Thank you @vsingh-msys! From a docs perspective, this PR should be good and thus approved, but it still needs review from @msorens before merging.

**Review by msorens — APPROVED** [review]
Testing steps working nominally now.

**Pull request merged**
